test/NuGet.Tests.Apex/NuGet.OptProf/NuGet.OptProf.csproj

```diff
@@ -22,11 +22,22 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Test.Apex.VisualStudio" />
+    <PackageReference Include="Microsoft.Test.Apex.VisualStudio" ExcludeAssets="Compile" GeneratePathProperty="true" />
     <PackageReference Include="Microsoft.VisualStudio.Sdk" />
+    <PackageReference Include="MSTest.TestAdapter" />
+    <PackageReference Include="MSTest.TestFramework" />
     <PackageReference Include="Newtonsoft.Json" NoWarn="NU1605" />
     <PackageReference Include="Xunit.StaFact" />
   </ItemGroup>
+  
+  <ItemGroup>
```

Comment:
I'm not necessarily excited about this change. 

Are the compiler warnings really worth referencing the assemblies in a custom way? Can we just no warn that instead?

Reply:
> I'm not necessarily excited about this change.

This one definitely feels like the least worst option 😄 

Here is what I considered:

1. The package has a terrible layout with an EXE in the `lib` folder and all of its runtime dependencies next to it
  ![image](https://user-images.githubusercontent.com/17556515/190496708-a42255e2-3037-4a3f-aea7-ed908f7cf243.png)
2. The package contains a `.targets` file that copies everything from the `tools` directory
  ```xml
<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <ItemGroup>
    <None Include="$(MSBuildThisFileDirectory)..\tools\*">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <Visible>false</Visible>
    </None>
  </ItemGroup>
</Project>
  ```
3.  The `tools` folder contains actual runtime dependencies
  ![image](https://user-images.githubusercontent.com/17556515/190497025-f09cd286-9376-430f-9ba4-440083056377.png)

At first I checked if a newer version of the package had a better layout, no dice.

Then I looked at suppressing the warning but the compiler warning is generic, the reference assembly is a different CPU, so suppressing the warning would apply to the entire project and all of its references.  If you know of a way to suppress this warning for a single assembly I'll gladly use that instead.  I would NoWarn the PackageReference instead but that doesn't work for this scenario.

I could write a custom target that _removes_ the references that cause the warning right after `ResolveAssemblyReferences` but I felt that would be harder to understand.

So I settled on manual consumption of the compile-time assets.  I'm open to other suggestions!

Reply:
I'll let others comment. 

Can we ask the owners to fix this, given that it's internal? 

We don't build apex tests locally often enough, so I'm not too worried about it as long it works.

Reply:
+1. Let's email the owners to fix the package. In the meantime, I think we can take these changes.